### ISSUE TITLE
onChange Input Focus for Multiple Inputs with Image Handler

### ISSUE THREAD

**Issue by hariria** [zenoamaro/react-quill#528]
# Problem -- related to #169 (Image Handler and input focus)

Thanks again so much for your work on react-quill. This is some pretty remarkable stuff so keep up the good work. <br/><br/>

I was following a different suggestion provided in issues #169 and recently ran into an issue where I was trying to add an image handler to embed images instead of uploading as a URL. I have another input box and for some reason, now whenever I add text to that input, it automatically changes the cursor to focus on ReactQuill. Seems extremely strange. If you could help me out I would really appreciate it!

# Codesandbox Demonstration: 
https://codesandbox.io/embed/confident-gagarin-qts90?fontsize=14

**Comment by jfradley** [comment]
Could be related to [this](https://github.com/zenoamaro/react-quill/issues/495) issue. There's an open [PR](https://github.com/zenoamaro/react-quill/pull/496) to resolve it if you wanted to test.

**Comment by hariria** [comment]
Thanks, @jfradley how would I npm install your forked version?

**Comment by hariria** [comment]
Ok, I just went into my `node_modules` and made the following edits in `src/component.js` according to the PR and it seemed to work fine without me having to rebuild:

`src/component.js - line 206`
```js
var QuillComponent = createClass({
		// Restore editor from Quill's native formats in regeneration scenario
		if (this.quillDelta) {
			this.editor.setContents(this.quillDelta);
			this.editor.setSelection(this.quillSelection);		
			this.editor.focus();
			this.quillDelta = this.quillSelection = null;
			if (this.quillSelection) {
				this.editor.setSelection(this.quillSelection);
			}
			if (this.quillHasFocus) {
				this.editor.focus();
			}
			this.quillDelta = this.quillSelection = this.quillHasFocus = null;
			return;
		}
		if (this.state.value) {
```
`src/component.js - line 321`
```js
var QuillComponent = createClass({
		// Cache selection and contents in Quill's native format to be restored later
		this.quillDelta = this.editor.getContents();
		this.quillSelection = this.editor.getSelection();
		this.quillHasFocus = this.editor.hasFocus();
		this.setState({
			generation: this.state.generation + 1,
		});
```

If the contributors could merge this PR I would really appreciate it. It's a pretty important and necessary change. I did notice that there was a slight bit of lag when I typed in my input box, but it's much better than before. Thanks @jfradley 

@zenoamaro (maintainer)
@alexkrolick (maintainer)
@clemmy
@asiniy
@webcarrot
@druti
@e-jigsaw
@zhang-z
@Sajam

**Comment by theodesp** [comment]
I just experienced the same issue suddenly. I have an input text above a ReactQuill component and when I type the imput the cursor gets focused on the editor instead. I dont have an image handler though:
```
<ReactQuill
          className={classnames('editor-content-area', {
            invalid: bodyState && !bodyState.valid,
            pristine: bodyState && bodyState.pristine,
          })}
          defaultValue={''}
          value={body}
          theme={null}
          modules={{
            toolbar: {
              container: '#toolbar',
            },
          }}
          children={<div data-quill-editor={true}/>}
          formats={[
            'bold', 'italic', /*'underline',*/ 'list', 'bullet', 'indent'
          ]}
          onFocus={handleBodyFocus}
          onChange={handleBodyChange}
          {...restProps}
        />
```

**Comment by hariria** [comment]
@theodesp yeah it seems like a pretty big issue. I'm not sure why it keeps on changing focus but it definitely needs to be fixed

**Comment by zehawki** [comment]
I had the same issue after adding imageHandler. Drove me nuts for a while, so I tried various things and here's what I'm using successfully:

The modules json is assigned as a state variable (in constructor) and then in ReactQuill I use `modules={this.state.modules}`, and this solved the issue. Note that I'm using class components.

**Comment by chetansgit** [comment]
Please refer to this link:

https://stackoverflow.com/questions/59825450/react-quill-custom-image-handler-module-causing-typing-issues-with-the-editor

**Comment by SanayRaj** [comment]
It's simple, Just make a new React Component file and make the quill things in it. and also make that component as a `React.PureComponent` or use `memo(*Your Component name goes here*)` hook

**Comment by Sukriti-sood** [comment]
I am facing same problem of cursor change, Is it solved now ? Or any solution
